### PR TITLE
 Format, document and debug

### PR DESCRIPTION
**Pull request by karosc**
I saw in the swmm-python issues you're planning on revising the error checking logic in your code. Consider merging these changes too. No functionality change yet, mostly just documenting and some minor refactoring to minimize outfile open vs close errors. I have written some pandas functions too, largely to pull timeseries for multiple links into a single dataframe with columns for datetime, link, and value. However, I will wait until the team sorts out how that should be handled. 

#### Format, document and debug
- Add RST docs everywhere
- Add type hinting everywhere
- Add element loading functions that are decorated with
   output_open_handler to avoid segmentation fault errors
   when accessing the outfile without opening it. Refactored
   these loading functions to use dict comprehension.
- Revise output_open_handler to use @wraps to sphinx works properly
- Move element attribute dicts into class properties and append
   pollutant names to appropriate dicts when opening an output file

### CODE REVIEW

**Comment by jennwuu** [comment]
Oh wow! I missed this PR when I was working through pyswmm PR comments. I would love to incorporate these awesome docstrings. Is it possible to rebase this PR to the latest master branch? @karosc 

I removed a lot of the redundant checks since the last commit. 

Let me know what I can do to help I would love to incorporate these changes. Thanks!

**Comment by jennwuu** [comment]
ping @karosc  - I want to follow up regarding the PR. I would love to include this into the branch but would need help fixing the merge conflict.

**Comment by karosc** [comment]
I think I resolved the conflicts well and fixed some errors on my part that I caught while developing swmm-pandas.

**Comment by jennwuu** [comment]
Awesome @karosc I will review through your changes. Happy holidays!

File: pyswmm/output.py
Comment (jennwuu) [inline pyswmm/output.py]: minor comment, can you adjust the spacing in this section?

**Review by jennwuu — APPROVED** [review]
@karosc These docstrings are so awesome! Just a minor comment on the spacing in one of the docstring.

**Comment by jennwuu** [comment]
Super super super super duper awesome! Love these changes! Thank you!

File: pyswmm/output.py
Comment (karosc) [reply pyswmm/output.py]: I fixed the indents in link_attribute and system_attribute doc strings

**Comment by jennwuu** [comment]
Awesome! Feel free to merge whenever. I approved the PR. Thanks!

**Comment by karosc** [comment]
I don't think I can merge since I don't have write access to your fork. Also github is showing 1 workflow is awaiting approval...

**Comment by jennwuu** [comment]
@karosc Great work! This is super awesome. I merged your PR.

**Pull request merged**
